File: LeetCode/longest_common_prefix.cpp
// https://leetcode.com/problems/longest-common-prefix/

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string comp = strs[0];
        for(int i=1; i<strs.size(); i++) {
            string temp = "";
            for(int j=0; j<strs[i].size(); j++) {
                if(comp[j] == strs[i][j]) {
                    temp += strs[i][j];
                } 
                else{
                    break;
                }
            }
            comp = temp;
        }
        return comp;
    }
};

File: LeetCode/longest_common_prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        comp = strs[0]
        for i in range(1, len(strs)):
            temp = ""
            for j in range(min(len(comp), len(strs[i]))):
                if comp[j] == strs[i][j]:
                    temp = temp + strs[i][j]
                else:
                    break
            comp = temp
        
        return comp